src/control/server/storage/config.go

```diff
@@ -572,12 +575,120 @@ func parsePCIBusRange(numRange string, bitSize int) (uint8, uint8, error) {
 	return uint8(begin), uint8(end), nil
 }
 
+// AccelOptionBits is a type alias representing optional capabilities as a bit-set.
+type AccelOptionBits uint16
+
+// toStrings returns a slice of option names that have been set.
+func (obs AccelOptionBits) toStrings() []string {
+	opts := common.NewStringSet()
+	for str, flag := range accelOptStr2Flag {
+		if obs&flag == flag {
+			opts.Add(str)
+		}
+	}
+
+	return opts.ToSlice()
+}
+
+// fromStrings generates bit-set referenced by the function receiver from the option names provided.
+func (obs *AccelOptionBits) fromStrings(opts ...string) error {
+	if obs == nil {
+		obs = new(AccelOptionBits)
+	}
+
+	for _, opt := range opts {
+		flag, exists := accelOptStr2Flag[opt]
+		if !exists {
+			return FaultBdevAccelOptionUnknown(opt, accelOptStr2Flag.keys()...)
+		}
+		*obs |= flag
+	}
+
+	return nil
+}
+
+func (obs AccelOptionBits) MarshalYAML() (interface{}, error) {
+	return obs.toStrings(), nil
+}
+
+func (obs *AccelOptionBits) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var opts []string
+	if err := unmarshal(&opts); err != nil {
+		return err
+	}
+
+	return obs.fromStrings(opts...)
+}
+
+// IsEmpty returns true if no options have been set.
+func (obs *AccelOptionBits) IsEmpty() bool {
+	return obs == nil || *obs == 0
+}
+
+// AccelProps struct describes acceleration engine setting and optional capabilities expressed
+// as a bit mask. AccelProps is used both in YAML server config and JSON NVMe config files.
+type AccelProps struct {
+	Engine  string          `yaml:"engine,omitempty" json:"accel_engine"`
+	Options AccelOptionBits `yaml:"options,omitempty" json:"accel_opts"`
+}
+
+type optFlagMap map[string]AccelOptionBits
+
+func (aosf optFlagMap) keys() []string {
+	keys := common.NewStringSet()
+	for k := range aosf {
+		keys.Add(k)
+	}
+
+	return keys.ToSlice()
+}
+
+var accelOptStr2Flag = optFlagMap{
+	accelOptCRCName:  AccelOptCRCFlag,
+	accelOptMoveName: AccelOptMoveFlag,
+}
+
+func (ap *AccelProps) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if ap == nil {
+		return nil
+	}
```

Comment:
I think that this could lead to confusing behavior. If `AccelProps` is nil, trying to unmarshal into it should at least result in an error, if not a segfault. For the record, the JSON package just segfaults if you give it a nil unmarshal receiver, which to me is fine and reasonable behavior.

Reply:
done